### DISCUSSION TITLE
How to multiply dimensions of all lines/points?

### DISCUSSION THREAD

**Discussion by makew0rld** [mozman/ezdxf#536] [Q&A] [answered]
I am using a software that ignores the units specified in the DXF file, and just interprets all numbers as being millimeters. But the DXF files I have use inches as the unit. How can I convert (by multiplying) all the measurements in the file to be correct for millimeters instead? Changing the dimensions of an entity seems easy enough, but I can't see how to iterate over all the entities in a document and change them.

Thank you!

**Comment by mozman** [comment]
This is not as trivial as you may think, this post tries to clarify the question:

- You need an input DXF file for another application which interprets a drawing unit as 1 mm
- Your DXF files are created by a drawing unit is 1 inch

If this assumptions are true, you have to scale the whole DXF document (model space) by factor 25.4 (25.4 mm = 1 inch). This can be done with _ezdxf_ to some extent, but it will mess up DIMENSION entities, paper space layouts, and possibly other structures that I am not yet thinking of.

**Reply by makew0rld** [reply]
> If this assumptions are true, you have to scale the whole DXF document (model space) by factor 25.4 (25.4 mm = 1 inch).

Yes, that's correct. What would the code be to do this? Thanks. I'm willing to try this even if it messes up.

**Comment by mozman** [comment, marked as answer]
```Python
import ezdxf
from ezdxf.math import Matrix44
from ezdxf import units

doc = ezdxf.readfile("inch.dxf")

INCH_TO_MM = 25.4
m = Matrix44.scale(sx=INCH_TO_MM, sy=INCH_TO_MM, sz=INCH_TO_MM)
for entity in doc.modelspace():
    try:  # not all entities are transformable
        entity.transform(m)
    except NotImplementedError:
        print(f"ignored entity {str(entity)}")

# Not important, just to store the correct information in the DXF header:
doc.header["$INSUNITS"] = units.MM
doc.header["$MEASUREMENT"] = 1  # for metric; 0 for imperial
doc.saveas("mm.dxf")

```

**Reply by makew0rld** [reply]
Thank you! I will try this out in a few days time and let you know if I had any issues.

**Reply by makew0rld** [reply]
This works! Thank you.
